packages/iterator-utilities/src/lib/slice.ts

```diff
@@ -0,0 +1,119 @@
+import { drop } from './drop';
+import { dropLast } from './dropLast';
+import { empty } from './empty';
+import type { IterableResolvable } from './from';
+import { toIntegerOrInfinityOrThrow } from './shared/_toIntegerOrInfinityOrThrow';
+import { take } from './take';
+import { takeLast } from './takeLast';
+
+/**
+ * Creates an iterable with the elements from the `start` index to the `end` index (exclusive).
```

Comment:
Specify that just like Array#slice this does not modify the original input

Reply:
This was misleading, it should be `Produces` instead of `Creates` because it consumes the iterable in most cases (only exceptions being `slice(iterable)`, `slice(iterable, 0)`, `slice(iterable, 1, 0)` (start >= end) and `slice(iterable, -2, -3)` (start >= end)).

Array#slice doesn't modify the input, but this one *may*.